src/DssSpell.t.sol

```diff
@@ -887,271 +912,27 @@ contract DssSpellTest is DssSpellTestBase {
-        // Check if users can stake
-        {
-            uint256 before = vm.snapshot();
-
-            uint256 stakingWad = 100_000 * WAD;
-            _giveTokens(address(usds), stakingWad);
-            usds.approve(address(rewards), stakingWad);
-            rewards.stake(stakingWad);
-            assertEq(rewards.balanceOf(address(this)), stakingWad, "testUsds01Rewards/rewards-invalid-staked-balance");
-
-            vm.revertTo(before);
-        }
-    }
-
-    function testVestedRewardsDistributionJob() public {
         _vote(address(spell));
         _scheduleWaitAndCast(address(spell));
         assertTrue(spell.done(), "TestError/spell-not-done");
 
-        SequencerLike seq                         = SequencerLike(addr.addr("CRON_SEQUENCER"));
-        CronJobLike job                           = CronJobLike(addr.addr("CRON_REWARDS_DIST_JOB"));
-        VestedRewardsDistributionLike distUsdsSky = VestedRewardsDistributionLike(addr.addr("REWARDS_DIST_USDS_SKY"));
-
-        uint256 blkt = block.timestamp;
-
-        // Since distribute() was called in the spell, initially the job is not due for execution...
-        {
-            // `workable()` is allowed to modify state in this job, so we need to make sure to revert its effects.
-            uint256 before = vm.snapshot();
-
-            // ... even 1 second before the desired interval, it would not be due
-            skip(7 days - 1 hours - 1);
-            // sanity check
-            assertFalse(blkt == block.timestamp);
-
-            (bool ok, ) = job.workable(seq.getMaster());
-            assertFalse(ok, "testVestedRewardsDistributionJob/unexpected-due-job");
-
-            vm.revertTo(before);
-        }
-
-        // sanity check: vm.revertTo() restores previous block.timestamp
-        assertTrue(blkt == block.timestamp);
-        // ... as we move to the next due date, the job can execute
-        skip(7 days - 1 hours);
-
-        {
-            // `workable()` is allowed to modify state in this job, so we need to make sure to revert its effects.
-            uint256 before = vm.snapshot();
-
-            (bool ok, bytes memory out) = job.workable(seq.getMaster());
-            assertTrue(ok, "testVestedRewardsDistributionJob/missing-due-job");
-            (address dist) = abi.decode(out, (address));
-            assertEq(dist, address(distUsdsSky), "testVestedRewardsDistributionJob/invalid-dist-returned");
-
-            vm.revertTo(before);
-        }
-
-        uint256 plastDistributed = distUsdsSky.lastDistributedAt();
-        job.work(seq.getMaster(), abi.encode(address(distUsdsSky)));
-        assertGt(distUsdsSky.lastDistributedAt(), plastDistributed, "testVestedRewardsDistributionJob/missing-distribution");
+        assertEq(split.flapper(), addr.addr("MCD_FLAP"), "testFlapperChange/incorrect-split-flapper-address");
+        assertEq(WardsAbstract(addr.addr("MCD_FLAP")).wards(addr.addr("MCD_SPLIT")),  1, "testFlapperChange/flapper-not-in-splitter-wards");
```

Comment:
For full coverage, we should also test the initialization, with at least the parameters included in the exec doc (want, pip, pair, usds, splitter).

Reply:
Also on previous spells for new components a full integration E2E test was included (i.e: mock a larger surplus and force a burn to see if nothing breaks or is misconfigured).

Reply:
There is `testSplitter` base test which tests flapper end-to-end. This is the test that matches your description "mock a larger surplus and force a burn to see if nothing breaks or is misconfigured". And there is `testFlapperChange`. More specifically:
- `want` is tested [here](https://github.com/makerdao/spells-mainnet/blob/14f22c1bb55ff52a8f60f9b398594969cebe4781/src/DssSpell.t.base.sol#L629-L632)
- `pip` is tested [here](https://github.com/makerdao/spells-mainnet/blob/14f22c1bb55ff52a8f60f9b398594969cebe4781/src/DssSpell.t.base.sol#L2553)
- `pair` is tested [here](https://github.com/makerdao/spells-mainnet/blob/14f22c1bb55ff52a8f60f9b398594969cebe4781/src/DssSpell.t.base.sol#L2554)
- `usds` address is actually only provided for sanity test and is not being set anywhere
- `splitter` [changes](https://github.com/makerdao/spells-mainnet/blob/14f22c1bb55ff52a8f60f9b398594969cebe4781/src/dependencies/dss-flappers/FlapperInit.sol#L129-L131) made by the init script are tested by the `testFlapperChange` (the two lines that you've commented on)